src/handlers/notebook.ts

```diff
@@ -63,6 +70,34 @@ export class NotebookHandler implements IDisposable {
     });
   }
 
+  private onCurrentFrameChanged(
+    callstackModel: Callstack.Model,
+    frame: Callstack.IFrame
+  ) {
+    const notebook = this.notebookTracker.currentWidget;
+    if (!notebook) {
+      return;
+    }
+
+    const cells = notebook.content.widgets;
+    cells.forEach(cell => CellManager.cleanupHighlight(cell));
+
+    if (!frame) {
+      return;
+    }
+
+    cells.forEach((cell, i) => {
+      // check the event is for the correct cell
+      const code = cell.model.value.text;
+      const cellId = this.debuggerService.getCellId(code);
```

Comment:
Attached properties would avoid recomputing the cell id (although this can be done later).
Also I wonder if it would be worth maintaining a mapping CellId <-> Cells so we don't have to for through all the cells when a frameChanged event is fired.

Reply:
> Attached properties would avoid recomputing the cell id (although this can be done later).

But the id / hash would still need to be computed when the cell changes.

> Also I wonder if it would be worth maintaining a mapping CellId <-> Cells so we don't have to for through all the cells when a frameChanged event is fired.

That sounds like a good idea. Do you think this is something we could do as part of #200?

Reply:
Yes.